packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2730,22 +2803,53 @@ function commitPassiveMountOnFiber(
       }
 
       if (enableTransitionTracing) {
-        if (committedTransitions !== null) {
+        // Get the transitions that were initiatized during the render
+        // and add a start transition callback for each of them
+        const state = finishedWork.memoizedState;
+        if (state.transitions === null) {
```

Comment:
Since it's a mutable field, this should probably live on the FiberRoot, too. Can do in a follow up.